### PR TITLE
encryption documentation updates

### PR DESCRIPTION
**Pull request by jrfastab**
Update documentation:
  * Add Getting started instructions for key updates
  * Add note that encrypt-interface is optional
  * Update Arch diagrams to include encryption block
  * Add L3 Encryption description to Arch section
  * Use "IPsec" instead of "IPSec" consistently

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/8726)

### CODE REVIEW

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
@aanm Any thoughts on this
```

   KEYID=$(kubectl get secret -n cilium cilium-ipsec-keys -o yaml|grep keys: | awk '{print $2}' | base64 -d | awk '{print $1}')
    if [[ $KEYID -gt 15 ]]; then KEYID=0; fi
    data=$(echo "{\"stringData\":{\"keys\":\"$((($KEYID+1))) "rfc4106\(gcm\(aes\)\)" $(echo $(dd if=/dev/u random count=20 bs=1 2> /dev/null| xxd -p -c 64)) 128\"}}")
    kubectl patch secret -n cilium cilium-ipsec-keys -p="${data}" -v=1
``` 

Only asking because you had some feedback on the original key generation one-liner.

File: Documentation/gettingstarted/encryption.rst [hunk 1]
```diff
@@ -167,6 +169,28 @@ exec -ti cilium-7cpsm -- bash`` and execute the following commands:
...
+============
+
+To replace cilium-ipsec-keys secret with a new keys,
+
+.. code-block:: shell-session
+
+    KEYID=$(kubectl get secret -n cilium cilium-ipsec-keys -o yaml|grep keys: | awk '{print $2}' | base64 -d | awk '{print $1}')
+    if [[ $KEYID -gt 15 ]]; then KEYID=0; fi
+    data=$(echo "{\"stringData\":{\"keys\":\"$((($KEYID+1))) "rfc4106\(gcm\(aes\)\)" $(echo $(dd if=/dev/urandom count=20 bs=1 2> /dev/null| xxd -p -c 64)) 128\"}}")
+    kubectl patch secret -n cilium cilium-ipsec-keys -p="${data}" -v=1
```
Comment (aanm) [inline Documentation/gettingstarted/encryption.rst:182]: Does this work as is? Don't you need the `op`, `path` of json patch?

Comment (jrfastab) [reply Documentation/gettingstarted/encryption.rst:182]: It worked when I tested it and verified by calling 'get secret' and checking cilium_ipcache map so seems sufficient.

**Pull request merged**